### PR TITLE
Fixes menu positioning on Wayland.

### PR DESCRIPTION
**Pull request by kbloom**
Change popup menus from Qt::Tool to Qt::Popup.
Thanks to Kai Uwe Broulik for pointing out the probable fix.

Resolves: #15285

- [X] I signed the [CLA](https://musescore.org/en/cla) (yes, I checked with my employer now, and it's OK)
- [X] The title of the PR describes the problem it addresses
- [X] Each commit's message describes its purpose and effects, and references the issue it resolves
- [X] If changes are extensive, there is a sequence of easily reviewable commits
- [X] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [X] There are no unnecessary changes
- [X] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Comment by cbjeukendrup** [comment]
I've investigated the problem on macOS identified here: https://github.com/musescore/MuseScore/pull/24094#issuecomment-2295843292
It doesn't look like a _logical_ solution for it is available. And on macOS there wasn't really any problem with Qt::Tool. So maybe we should just not apply this change on macOS. That can be done like this:
```c++
#ifdef Q_OS_MAC
            Qt::Tool
#else
            Qt::Popup
#endif
```

**Comment by kbloom** [comment]
Done

**Comment by cbjeukendrup** [comment]
Thanks! Then this will need testing only on Windows and Linux, @DmitryArefiev (in this version of the PR, nothing changes for macOS). Also keep in mind Roman's comment here: https://github.com/musescore/MuseScore/pull/24094#issuecomment-2295843292

**Comment by kbloom** [comment]
I tested tooltips and linux and they're also centered if they're Qt::Tool, and seem to work normally if they're Qt::Popup.

**Comment by MarcSabatella** [comment]
I'm happy to test, but it would be nice to have a build that also includes (or excludes) the necessary library so that it runs at all.

File: src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp
Comment (RomanPudashkin) [inline src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp:78]: @cbjeukendrup do we really need this change on windows? It works perfectly fine there. So it should be:

```
#if defined(Q_OS_MAC) || defined(Q_OS_WIN)
    Qt::Tool
#else
    Qt::Popup // Popups can't be Qt::Tool on Linux Wayland, or they can't be relatvely positioned.
#endif
```

**Comment by kbloom** [comment]
Would you like me to rebase this on top of #24111 before that PR is merged, or should I wait until it's merged?

**Comment by cbjeukendrup** [comment]
I think we should be able to merge #24111 very soon, so if you wait until after that, you can just rebase onto latest master. Also while doing that, it's a good idea to apply Roman's suggestion too (sorry that I didn't include that in my suggestion about macOS). Not unnecessarily making changes on Windows too reduces the risk and testing load of this PR.

**Comment by DmitryArefiev** [comment]
> Thanks! Then this will need testing only on Windows and Linux, @DmitryArefiev (in this version of the PR, nothing changes for macOS). Also keep in mind Roman's comment here: [#24094 (comment)](https://github.com/musescore/MuseScore/pull/24094#issuecomment-2295843292)

Tested on **LinuxUbuntu22.04.4 LTS**. **No regressions found** comparing with master  

But I was not able to reproduce #15285 previously

**Comment by cbjeukendrup** [comment]
Great! I will create a 4.4 version.

**Comment by kbloom** [comment]
Changed to use a #ifdef Q_OS_LINUX, so that this doesn't change anything on windows (where this Qt::Tool already seems to work.)

**Review by cbjeukendrup — APPROVED** [review]

**Pull request merged**
